repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,274 @@
+    except OSError:  # pragma: no cover
+        console.print(f"Failed to download initial root from {root_url}")
+        return False  # pragma: no cover
+
+    console.print(
+        f"Trust-on-First-Use: Initialized new root in {metadata_dir}"
+    )
+    return True
+
+
+def perform_tuf_ngclient_download_artifact(
+    metadata_url: str,
+    metadata_dir: str,
+    artifacts_url: str,
+    artifact_name: str,
+    download_dir: str,
+    config: UpdaterConfig,
+) -> bool:
+    try:
+        updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=metadata_url,
+            target_base_url=artifacts_url,
+            target_dir=download_dir,
+            config=config,
+        )
+        updater.refresh()
+
+        # Annotating, as it's an external method we don't want to test
+        info = updater.get_targetinfo(artifact_name)  # pragma: no cover
+
+        if info is None:  # pragma: no cover
+            console.print(f"Artifact {artifact_name} not found")
+            return False
+
+        path = updater.find_cached_target(info)  # pragma: no cover
+        if path:  # pragma: no cover
+            console.print(f"Artifact is available in {path}")
+            return True
+
+        path = updater.download_target(info)  # pragma: no cover
+        console.print(  # pragma: no cover
+            f"Artifact downloaded and available in {path}"
+        )
+
+    except (OSError, RepositoryError, DownloadError) as e:
+        console.print(f"Failed to download artifact {artifact_name}: {e}")
+        return False
+
+    return True
+
+
+def _download_artifact(
+    metadata_url: Optional[str],
+    artifacts_url: Optional[str],
+    hash_prefix: Optional[bool],
+    directory_prefix: Optional[str],
+    artifact_name: str,
+    root: Optional[str],
+) -> bool:
+    if metadata_url is None:
+        console.print("Please specify metadata url")
+        return False
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    if artifacts_url is None:
+        console.print("Please specify artifacts url")
+        return False
+
+    if not os.path.isfile(f"{metadata_dir}/root.json"):
+        console.print(
+            "Trusted local root not found. Using 'tofu' to "
+            "Trust-On-First-Use or copy trusted root metadata to "
+            f"{metadata_dir}/root.json"
+        )
+        ok = init_tofu(metadata_url, root)
+        if not ok:
+            console.print(
+                f"Trusted local root not found in {metadata_url} - "
+                "`tofu` was not successful"
+            )
+            return False
+
+    print(f"Using trusted root in {metadata_dir}")
+
+    if directory_prefix:
+        # Annotating as it's covered, but codecov doesn't
+        # recognize that properly
+        download_dir = directory_prefix  # pragma: no cover
+    else:
+        download_dir = os.getcwd() + "/downloads"
+
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
```

Comment:
```suggestion
        os.mkdir(download_dir, exist_ok=True)
```

Reply:
Ah, this parameter is still not fixed and causing issues ever since some of the 3.4 versions till now. Now I remember why I removed it.

Breaking tax locally and in the CI. I've put an ignore annotation. If you think this hides risk of unexpected issues at some point, I'd just remove it.

Reply:
I removed the change, it's still breaking the CI.